src/context.ts

```diff
@@ -0,0 +1,28 @@
+import { prisma, Prisma, User } from "./generated/prisma-client";
+import Cookies from "universal-cookie";
+import { resolveUserUsingJWT } from "./utils/resolveUser";
+
+export type AppContext = {
+  prisma: Prisma;
+  token: string;
+  auth: {
+    user: User;
+    isAuthenticated: boolean;
+  };
+};
+
+export const context = async ({ req }): Promise<AppContext> => {
```

Comment:
maybe should rename this function as `createContext`

Reply:
I did give this a good thought. Although before I give my opinions, I would like to know why you say that?

Reply:
I don’t know. This function creates and returns a context. So what do you think?

Reply:
Also there’s `AppContext` being exported. Imagine that.

Reply:
I see.

I will put this differently. At first glance, it does look like a function that is returning a context. However, it is doing something different.

Think of what the context type is in the the Apollo server docs:
https://github.com/apollographql/apollo-server/blob/71361880b6ba284ee8d16d5091945a999b005703/packages/apollo-server-express/src/ApolloServer.ts#L83

It says `ContextFunction<ExpressContext, Context> | Context`. What we do here is pass a context function as our context.

So it is not returning context, it is the context. Hope this explains things enough.

Reply:
I see. I was thinking about this too.